editor.planx.uk/src/pages/Preview/Questions.tsx

```diff
@@ -69,59 +69,66 @@ const Questions = ({ previewEnvironment }: QuestionsProps) => {
   const isStandalone = previewEnvironment === "standalone";
   const { createAnalytics, node } = useAnalyticsTracking();
   const [gotFlow, setGotFlow] = useState(false);
-  const isSingleSession =
+  const isUsingLocalStorage =
     useStore((state) => state.path) === ApplicationPath.SingleSession;
 
-  if (isSingleSession) {
-    // Use local storage for simple, non-Save&Return flows
-    useEffect(() => {
-      setPreviewEnvironment(previewEnvironment);
-      if (isStandalone) {
-        const state = getLocalFlow(id);
-        if (state) {
-          resumeSession(state);
-        }
+  useEffect(
+    () => setPreviewEnvironment(previewEnvironment),
+    [previewEnvironment, setPreviewEnvironment],
+  );
+
+  // Initial setup
+  useEffect(() => {
+    if (!isStandalone) return;
+
+    if (isUsingLocalStorage) {
+      const state = getLocalFlow(id);
+      if (state) resumeSession(state);
+      createAnalytics(state ? "resume" : "init");
+      setGotFlow(true);
+    } else {
+      NEW.getLocalFlow(sessionId).then((state) => {
+        // session data is resumed by ./ResumePage.tsx
         createAnalytics(state ? "resume" : "init");
         setGotFlow(true);
-      }
-    }, []);
-
-    useEffect(() => {
-      if (gotFlow && isStandalone && id) {
-        setLocalFlow(id, {
-          breadcrumbs,
-          id,
-          passport,
-          sessionId,
-          govUkPayment,
-        });
-      }
-    }, [gotFlow, breadcrumbs, passport, sessionId, id, govUkPayment]);
-  } else {
-    // Use lowcalStorage for Save & Return flows
-    useEffect(() => {
-      setPreviewEnvironment(previewEnvironment);
-      if (isStandalone) {
-        NEW.getLocalFlow(sessionId).then((state) => {
-          // session data is resumed by ./ResumePage.tsx
-          createAnalytics(state ? "resume" : "init");
-          setGotFlow(true);
-        });
-      }
-    }, []);
-
-    useEffect(() => {
-      if (gotFlow && isStandalone && sessionId) {
-        NEW.setLocalFlow(sessionId, {
-          breadcrumbs,
-          id,
-          passport,
-          sessionId,
-          govUkPayment,
-        });
-      }
-    }, [gotFlow, breadcrumbs, passport, sessionId, id, govUkPayment]);
-  }
+      });
+    }
+    // Updating useContext (createAnalytics) within useEffect leads to recursive calls
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    // createAnalytics
```

Comment:
I spent a good while trying to resolve this one without much luck.

`createAnalytics()` comes from the `useContext()` hook, and nesting them exceeds maximum call depth when hitting "Continue". Presumably this is because a reference for the context changes?

This was intended as a quick-fix PR whilst I waited on IaC to complete something and this issue de-railed this 😅 🚂 

I think what's here is an improvement and more readable, and I'll come back and have a pass at this one another time if that works?